src/ast/values.rs

```diff
@@ -165,6 +170,33 @@ impl<'a> ParameterizedValue<'a> {
     pub fn into_string(self) -> Option<String> {
         match self {
             ParameterizedValue::Text(cow) => Some(cow.into_owned()),
+            ParameterizedValue::Bytes(cow) => String::from_utf8(cow.into_owned()).ok(),
+            _ => None,
+        }
+    }
+
+    /// Returns whether this value is the `Bytes` variant.
+    pub fn is_bytes(&self) -> bool {
+        match self {
+            ParameterizedValue::Bytes(_) => true,
+            _ => false,
+        }
+    }
+
+    /// Returns a bytes slice if the value is text or a byte slice, otherwise `None`.
+    pub fn as_bytes(&self) -> Option<&[u8]> {
+        match self {
+            ParameterizedValue::Text(cow) => Some(cow.as_ref().as_bytes()),
+            ParameterizedValue::Bytes(cow) => Some(cow.as_ref()),
+            _ => None,
+        }
+    }
+
+    /// Returns a cloned `Vec<u8>` if the value is text or a byte slice, otherwise `None`.
+    pub fn to_bytes(&self) -> Option<Vec<u8>> {
```

Comment:
`into_bytes`

Reply:
by convention `into_bytes` would take ownership though, right? I think there's also an `into_bytes` already. If not I'll add it.

Reply:
Achso. Yes.